src/validation.cpp

```diff
@@ -3817,6 +3840,13 @@ bool static LoadBlockIndexDB(const CChainParams& chainparams)
 
     boost::this_thread::interruption_point();
 
+    // build mapPrevBlockIndex
+    for (auto& p : mapBlockIndex) {
+        if (p.second->pprev) {
```

Comment:
could avoid looping through `mapBlockIndex` twice if you would move this into `BOOST_FOREACH` which fills `vSortedByHeight` below